js/lib/urlutil.js

```diff
@@ -429,8 +430,11 @@ const UrlUtil = {
     }
 
     let parsed = urlParse(location)
-    if (parsed.origin) {
-      // parsed.origin is specific to muon.url.parse
+    // parsed.origin is specific to muon.url.parse
+    if (parsed.origin !== undefined) {
+      if (parsed.protocol === 'about:') {
```

Comment:
on second thought, using the path as the hostname as a generic fallback doesn't seem very safe. so i am fine with just having an exception for 'about:' for now.